templates/controller-deployment.yaml

```diff
@@ -62,10 +55,14 @@ spec:
             {{- toYaml .Values.securityContext | nindent 12 }}
           image: "{{ .Values.image.repository }}:{{ .Values.image.tag | default .Chart.AppVersion }}"
           imagePullPolicy: {{ .Values.image.pullPolicy }}
-          command: ["/bin/boundary", "server"]
+          command: ["/bin/sh", "-ec" ]
           args:
-            - "-config"
-            - "/boundary/boundary-config.hcl"
+            - |
+              {{- if .Values.vault.enabled }}
+              export VAULT_TOKEN=$(cat /vault/secrets/token)
```

Comment:
It's unclear why the Vault token needs to be exported like this. The Vault Agent sidecar runs and handles all the authentication to retrieve the secrets and expose the path for them to Boundary. Boundary still requires the token to be configured for further use through the Boundary Admin from all details I've compiled from docs and the Discuss forum. The sidecar also uses K8s authentication with Vault for the given `role`.

Reply:
Yes i know, the goal is to run boundary without a long time running sidecar (less ressources, easier to managed, etc...).  
vault-agent is used as an init container to retrieve a valid token and pass to boundary that will used and renew it.  
Also the vault token is pass to boundary as an environment variable and doesn't need to stay on disk (better for security reason).

Reply:
The vault-agent sidecar doesn't add that much in the way of resources, and if you're operating as I am using Vault dynamic database credentials you actually need to have that sidecar running as it will uptime the credentials.

Reply:
Hi, i retrieve database credentials from vault and the vault-agent is used to get the credentials but only on the start of the container.  
Can you try my branch with your current helm values to check ?

Reply:
I'm going to try and execute it myself when I've got a moment, but currently, I'm buried with work for my day job. I'm also looking to try and automate some testing as well.

Reply:
No problem. This MR is to use boundary with all it's functionality without relying on other process.